### ISSUE TITLE
RESEARCH: Other countries

### ISSUE THREAD

**Issue by tordans** [osmberlin/osm-traffic-sign-tool#11]
Let's collect know how on how this tool might be expanded to other countries…

(Please read also https://github.com/osmberlin/osm-traffic-sign-tool/issues/40 about the status of this project.)

# Link

* https://wiki.openstreetmap.org/wiki/DE:Key:traffic_sign#Werte
* Lists are at https://wiki.openstreetmap.org/wiki/DE:Key:traffic_sign#Liste_der_Kennungen_(IDs)_nach_Staat
* A list of existing preset definitions (for JOSM) https://wiki.openstreetmap.org/wiki/Key:traffic_sign#Presets,_styles,_libraries,_statistics_and_maps_with_traffic_signs_with_IDs_per_country

**Comment by eMerzh** [comment]
It would be really interesting indeed... i like the idea of having that for belgium 

there's already a list in here 
https://wiki.openstreetmap.org/wiki/Road_signs_in_Belgium (with sub pages of course)

**Comment by 1ec5** [comment]
Localization would be helpful for international expansion. The wiki’s sign catalog for the United States is [partially](https://wiki.openstreetmap.org/wiki/ES:MUTCD/Puerto_Rico) available in two languages, and the [Canadian](https://wiki.openstreetmap.org/wiki/Canada/Road_signs) community has indicated plans to do likewise.

**Comment by tordans** [comment]
Great to hear it @eMerzh, @1ec5. I saw that Weekly OSM picket up this project so [I wrote a "status of this project" issue](https://github.com/osmberlin/osm-traffic-sign-tool/issues/40) today to clarify the current status and how it might continue at some point. 

It would be great expand this to Belgium, US, Canada at some point!

**Comment by tordans** [comment]
Poland has a nice list of signs and tags as well https://wiki.openstreetmap.org/wiki/Pl:Road_signs_in_Poland

(And some fun signs to add … https://en.osm.town/@tordans/112557035738081302)

**Comment by llongour** [comment]
And see here for France : https://wiki.openstreetmap.org/wiki/FR:Signalisation_routi%C3%A8re_en_France 😄

**Comment by tordans** [comment]
Hey, does anyone here have the time to start adding a few traffic signs for your local language to the project? I can help create the basic setup for a second language and would like to use this opportunity to see where the whole setup still needs improvements. But I won't have the time to actually add all the signs, image files and so on…

**Comment by eMerzh** [comment]
Yeah, similar here, not 100% sure the extent of the task, but I can clearly help doing it :)

**Comment by eMerzh** [comment]
@tordans, i'm starting to create packages/internal_wiki/src/data/trafficSignsWiki.json for belgium... 
it's that what you need, right?

(btw: what exactly is the packageImageImportName? for the moment i make it a js importable name of the traffic sign 🤷 )

**Comment by tordans** [comment]
> @tordans, i'm starting to create packages/internal_wiki/src/data/trafficSignsWiki.json for belgium... 
> 
> it's that what you need, right?

It is https://github.com/osmberlin/osm-traffic-sign-tool/blob/main/packages/traffic-sign-converter/src/data/trafficSignData.ts

But better to wait for a few weeks because some issues came up that will likely end up in changes in the schema.

**Comment by eMerzh** [comment]
ok 👌  i'll wait then, let us know when you are ready

**Comment by andrewharvey** [comment]
If you're accepting contributions for other regions, I'm happy to help contribute for Australia based on https://wiki.openstreetmap.org/wiki/Australian_Tagging_Guidelines/Road_Signage

**Comment by tordans** [comment]
Side note: I just learned of this prototype integration for Australia https://github.com/osmberlin/osm-traffic-sign-tool/issues/84#issuecomment-2680416535

**Comment by matheusgomesms** [comment]
I am working on something like this related to a research: conflate OSM and Mapillary data.

This made me do some tables between Mapillary traffic signs and OSM tagging schema (not public yet). I also had to update this page regarding Brazil: https://wiki.openstreetmap.org/wiki/Pt:Placas_de_sinalização_no_Brasil

My idea is to work with traffic signs from Brazil, Portugal and Germany. Just did Brazil as a PoC, will work on the other countries in the following weeks.

Brazil is interesting because officially it doesn't have many signs (around 100). Let me know when this is ready to accept other countries, I would love to help.

**Comment by tordans** [comment]
@matheusgomesms thanks for sharing Brazil!

> conflate OSM and Mapillary data

That sounds interesting but also hard to do (in terms of getting this actionable for mapping). Would love to learn more, maybe in Slack US Chat?

Would it be of help if we added the Mapillary sign id to the traffic sign data in the tool? You could use that as a lookup for tag suggestions and the official sign ID.

Looking at the list there will be quite a few that the tool does not have, yet, because they are of low priority for mapping signs (like narrow road and such).

(Strings at https://www.mapillary.com/developer/api-documentation/traffic-signs and SVGs as https://github.com/mapillary/mapillary_sprite_source/tree/master/package_signs (at least until 2021))


> Let me know when this is ready to accept other countries, I would love to help.

I could set something up now (which is #100 ) but I am hesitant because looking at some of the issues the tool still has it is very likely I will have to change the data structure to resolve those which will be harder when more countries are present. That being said, a second country should not be a big issue… — Which country would you like to see added first?

**Comment by matheusgomesms** [comment]
> That sounds interesting but also hard to do (in terms of getting this actionable for mapping). Would love to learn more, maybe in Slack US Chat?

Sure, let me polish a bit my tool and I'll release to public discussion. My initial idea was to use this equivalence table data to be used everywhere, for example, in iD/JOSM, where you can display Mapillary items, but can't use them directly to mapping (one-click action, for example).

So more of Mapillary -> OSM, instead of the opposite direction (don't know, didn't think much about it yet!).



> I could set something up now (which is [#100](https://github.com/osmberlin/osm-traffic-sign-tool/issues/100) ) but I am hesitant because looking at some of the issues the tool still has it is very likely I will have to change the data structure to resolve those which will be harder when more countries are present. That being said, a second country should not be a big issue… — Which country would you like to see added first?

I would say Brazil because I'm working on that, and the list is fairly small I believe (compared to other countries), so a nice option to test the tool. But not in a hurry, I agree these issues should be sorted out first.

However, If you still want to release a beta with another country and see where the tool struggles, count with me, I can be your beta-tester and discuss where the tool is not working as it should or where should be improved :)

**Comment by tordans** [comment]
**I have a big update to share and ask for your help:**
The tool now has country catalogues for most of the countries mentioned in this thread.
There are QA Tools in place (Footer) that help to review the tagging suggestions and compare the tools catalogue against Taginfo and Wiki. It would be great if you used those to verify the data and create Github issues using the button on the page. When all works as planned, those Issues will be prefilled with information that will allow Cursor cloud to suggest PRs automatically for those changes and we can review them in a branch preview right here. – At least that is the plan :).

Check it out at https://trafficsigns.osm-verkehrswende.org/ the list of catalogues (and in the top right Lang-Switcher).
* https://trafficsigns.osm-verkehrswende.org/AU
* https://trafficsigns.osm-verkehrswende.org/AT
* https://trafficsigns.osm-verkehrswende.org/BE
* https://trafficsigns.osm-verkehrswende.org/BR
* https://trafficsigns.osm-verkehrswende.org/CA
* https://trafficsigns.osm-verkehrswende.org/FR
* https://trafficsigns.osm-verkehrswende.org/PL

And of course https://trafficsigns.osm-verkehrswende.org/DE

The UI Language is either EN or DE (see https://github.com/osmberlin/osm-traffic-sign-tool/issues/29#issuecomment-4641457485).
